### ISSUE TITLE
Updates to nomination review system posting conditions

### ISSUE THREAD

**Issue by wookie184** [python-discord/bot#1746]
## Updates to nomination review system posting conditions
There are a few changes we'd like to make to the nominations review system.

### Features we want:
1) A limit to the maximum number of reviews active in the channel at once.
2) A limit on the frequency reviews are posted (e.g. one a day).
3) A check to ensure the user has been active recently, to hold off the review if they haven't and possibly unnominate after a while.

#### General Idea:
These changes don't fit in that easily with the current scheduler based system, so I propose adding a second stage using an "awaiting review" status. So once their 30 days in the scheduler is up, they will be added to this "queue".

We could then have a `discord.ext.tasks.loop` run, maybe every hour, which checks a) that it's been 24 hours since the last vote; and b) that the number of active votes is below the maximum. If these are true then it will try and pick a user to review.

To pick a user, we'll loop over users awaiting review (probably ordered by first nominated) until we find one with recent activity, and then post their review.

#### Details:
"awaiting review" will probably need to be a new field stored on the nomination in postgres, or it may be better to change the current "reviewed" field to have multiple values, "not reviewed", "awaiting review", "being reviewed" (for keeping track of number of active reviews), and "reviewed".

For keeping track of the number of active votes, the message ID(s) of each vote could be stored on the nomination in postgres. When a user is unnominated the bot can use this to automatically delete the message(s) and archive it. 

Storing the message IDs with the user would help for archiving the vote so we have the information we need rather than trying to parse the messages with dodgy regexes to work out which review a message is from.

To handle reviews posted manually, we could replace manual posting with a bot command where you pass the user and the content you want the bot to paste. We *could* try and work out a way for this manual posting to handle multiple messages, but I think it would be easier to accept one and further content can go in a thread (although it wouldn't be archived).

The number of active votes can be calculated using the number of reviews with a "being reviewed" status.
 ___
I would be happy to try implementing these changes if the implementation I have suggested sounds OK, although it would be good if somebody else could handle the site side of things.

Also, feel free to suggest alternative implementations / point out flaws in what i've said, and let me know if i've missed out any details.

**Comment by wookie184** [comment]
Thinking about this a bit more, it would probably make more sense to remove the current scheduler completely and add another check that they've been in the pool long enough along with the check for recent activity.

**Comment by ChrisLovering** [comment]
I agree we should remove the current scheduler entirely. The hourly task you mention can then run the two checks you mention, and if they pass it should just make an API request to the site api to get the next one to review, which then does the required calcs to work out which user to return (as the site has all the message history & nomination details). No need for a scheduler at all then.

**Comment by MarkKoz** [comment]
>  A check to ensure the user has been active recently, to hold off the review if they haven't and possibly unnominate after a while.

This may seem like an innocent thing to add, but how do you propose to actually track user activity. Doesn't that also get into privacy issues?

**Comment by wookie184** [comment]
> This may seem like an innocent thing to add, but how do you propose to actually track user activity. Doesn't that also get into privacy issues?

By user activity I mean something like "has at least 10 messages in the past week", which we already track through [metricity](https://github.com/python-discord/metricity), which is integrated with site https://github.com/python-discord/site/blob/d038076b5034d14ce2b8dabcb929eacb0e837540/pydis_site/apps/api/models/bot/metricity.py, so would just require another query there.

**Comment by wookie184** [comment]
I'm going to work on a slightly simplified version of this.
